File: BeautifulStrings/cpp/BeautifulStrings.cpp
// Beautiful Strings solution in C++ for CodeEval.com by Steven A. Dunn

#include <algorithm>
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

int main(int argc, char* argv[])
{
	ifstream inputFile(argv[1]);
	string line;
	
	if (inputFile)
	{
		while (getline(inputFile, line))
		{
			const int ASCII_CONVERSION_VALUE = 97;
			vector<int> counts(26, 0);
			transform(line.begin(), line.end(), line.begin(), ::tolower);
			for (int strIdx = 0; strIdx < line.length(); strIdx++)
			{
				if (line[strIdx] - ASCII_CONVERSION_VALUE >= 0 && 
					line[strIdx] - ASCII_CONVERSION_VALUE <= 25)
				{  
					counts[line[strIdx] - ASCII_CONVERSION_VALUE]++;
				}
			}
			sort(counts.begin(), counts.end());
			
			int beauty = 0;
			int weight = 26;
			for (int countsIdx = counts.size(); countsIdx >= 0; countsIdx--)
			{
				beauty += (weight * counts[countsIdx-1]);
				weight--;
			}
			cout << beauty << endl;
		}
	}
}

File: BeautifulStrings/c#/BeautifulStrings.cs
// Beautiful Strings soln in c# for code eval by steven a dunn

using System;
using System.IO;

class BeautifulStrings {
  public static void Main(string[] args) {
    using (StreamReader reader = new StreamReader(args[0])) {
      while (!reader.EndOfStream) {
        string line = reader.ReadLine();
        if (null == line)
          continue;

        line = line.ToLower();

        int[] counts = getCounts(line);
        int beauty = getBeauty(counts);

        Console.WriteLine(beauty);
      }
    }
  }

  static int[] getCounts(string line) {
    int[] counts = new int[26];
    const int ASCII_ZERO = 97;

    foreach (char c in line) {
      int i = (int)c - ASCII_ZERO;
      if (i >= 0 && i < 26)
        counts[i]++;
    }
    return counts;
  }

  static int getBeauty(int[] counts) {
    int beauty = 0;
    int weight = 26;
 
    Array.Sort(counts);

    for (int i = counts.Length - 1; i >= 0; i--) {
      beauty += weight * counts[i];
      weight--;
    }

    return beauty;
  }
}